// Proto file representing the EDB SimSetupInfo class

syntax = "proto3";

package ansys.api.edb.v1;

import "edb_messages.proto";

// DEPRECATED: use SimulationSetupService instead
service SimulationSetupInfoService {
  // Gets the SimulationSettings of the SimulationSetupInfo
  rpc GetSimulationSettings (EDBObjMessage) returns (EDBObjMessage) {}

  // Gets the list of SweepData
  rpc GetSweepDataList (EDBObjMessage) returns (SweepDataListMessage) {}

  // Sets the list of SweepData
  rpc SetSweepDataList (SetSweepDataListMessage) returns (google.protobuf.Empty) {}
}

message SweepDataMessage {
  string name = 1;
  string frequency_string = 2;
  bool fast_sweep = 3;
}

message SweepDataListMessage {
  repeated SweepDataMessage sweep_data = 1;
}
 
message SetSweepDataListMessage {
  EDBObjMessage simulation_setup_info = 1;
  SweepDataListMessage sweep_data_list = 2;
}
